server/src/main/scala/services/Optimiser.scala

```diff
@@ -83,6 +83,7 @@ object Optimiser {
                   capacity: IndexedSeq[Int],
                   sla: Int,
                   qstart: IndexedSeq[Double]): ProcessedWork = {
+    assert(capacity.length == work.length, s"capacity & work don't match: ${capacity.length} vs ${work.length}")
```

Comment:
will this kill the app if it fails?

Reply:
No it just logs an error message and throws an exception

Reply:
I'll change it to an exception. I'd blindly recreated the assertions that are in the original R code, but an exception is presumably better practice